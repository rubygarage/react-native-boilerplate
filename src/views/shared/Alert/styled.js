import styled from 'styled-components/native';

import Icon from 'views/shared/Icon';

const Wrapper = styled.View`
  position: relative;
  flex-direction: row;
  margin-bottom: 8;
  padding-top: 12;
  padding-right: ${({ onCloseClick }) => (onCloseClick ? 48 : 12)};
  padding-bottom: 12;
  padding-left: 12;
  border-radius: 4;
  background-color: ${({ color }) => color};
`;

const IconWrap = styled.View`
  margin-right: 12;
  flex-shrink: 0;
`;

const ContentArea = styled.View`
  flex-grow: 1;
  padding-right: 36;
  display: flex;
  align-items: flex-start;
`;

const MessageTitle = styled.Text`
  flex-direction: row;
  flex-wrap: wrap;
  margin-top: 2;
  font-size: 14;
  line-height: 20;
  font-weight: bold;
  color: ${({ color }) => color};
`;

const MessageText = styled.Text`
  margin-bottom: 2;
  font-size: 14;
  line-height: 20;
  color: ${({ color }) => color};
`;

const ActionButton = styled.TouchableOpacity`
  flex-direction: row;
  align-items: center;
  max-width: 200;
  height: 36;
  margin-top: 8;
  padding-right: 12;
  padding-left: 12;
  border-radius: 4;
  background-color: ${({ color }) => color};
`;

const ActionIcon = styled(Icon)`
  margin-right: 24;
`;

const ActionButtonText = styled.Text`
  color: ${({ theme }) => theme.colors.white};
  font-size: 14;
  line-height: 20;
  font-weight: bold;
  letter-spacing: -0.2;
`;

const CloseButton = styled.TouchableOpacity`
  position: absolute;
  top: 12;
  right: 12;
  color: ${({ theme }) => theme.colors.white};
`;

export {
  Wrapper,
  IconWrap,
  ContentArea,
  MessageTitle,
  MessageText,
  ActionButton,
  ActionButtonText,
  CloseButton,
  ActionIcon,
};
