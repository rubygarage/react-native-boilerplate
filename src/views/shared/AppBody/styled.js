import styled from 'styled-components/native';

const AppBody = styled.View`
  position: relative;
  flex-grow: 1;
  padding-right: 16;
  padding-left: 16;
  background-color: ${({ theme }) => theme.colors.black};
`;

export default AppBody;
