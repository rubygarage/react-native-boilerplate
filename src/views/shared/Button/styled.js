import styled from 'styled-components/native';

const Wrap = styled.TouchableOpacity`
  min-width: ${(props) => (props.isSmallWidth ? 72 : 136)};
  height: ${(props) => (props.isSmallHeight ? '36px' : '48px')};
  justify-content: center;
  align-items: center;
  padding-right: 8;
  padding-left: 8;
  border-radius: 4;
  flex-shrink: 0;
  background-color: ${({ theme, isOutline, fillColor }) => (isOutline ? theme.colors.white : fillColor)};
  margin-right: ${(props) => (props.isMarginRight ? '16' : '0')};
  margin-bottom: ${(props) => (props.isMarginBottom ? '16' : '0')};
  border-width: ${(props) => (props.isBorder ? '2' : '0')};
  border-style: solid;
  border-color: ${({ theme }) => theme.colors.black};
`;

const Text = styled.Text`
  color: ${({ theme, isOutline }) => (isOutline ? theme.colors.black : theme.colors.white)};
  font-weight: bold;
`;

export { Wrap, Text };
