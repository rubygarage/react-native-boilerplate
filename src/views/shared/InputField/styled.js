import styled from 'styled-components/native';

const Wrap = styled.View`
  flex-direction: column;
`;

const Label = styled.Text`
  margin-bottom: 4;
  font-size: 14;
  line-height: 20;
  color: ${({ theme }) => theme.colors.black};
`;

const FieldWrap = styled.View`
  position: relative;
`;

const Field = styled.TextInput`
  border-radius: 3;
  min-height: 48;
  border-width: ${({ isFocused, isError }) => (isFocused && !isError ? 2 : 1)};
  border-style: solid;
  border-color: ${({ theme, isError, isFocused }) => {
    if (isError) {
      return theme.colors.black;
    }
    if (isFocused) {
      return theme.colors.black;
    }
    return theme.colors.black;
  }};
  padding-top: 10;
  padding-right: ${({ actionWidth }) => (actionWidth + 12)};
  padding-bottom: 10;
  padding-left: 12;
  font-size: 14;
  line-height: 20;
  color: ${({ theme }) => theme.colors.black};
`;

const FieldError = styled.View`
  flex-direction: row;
  align-items: center;
  margin-top: 6;
  margin-bottom: 8;
`;

const FieldErrorClose = styled.View`
  margin-right: 8;
`;

const FieldErrorMessage = styled.Text`
  color: ${({ theme }) => theme.colors.black};
  font-size: 14;
  line-height: 20;
`;

const ActionButtonWrapper = styled.View`
  position: absolute;
  height: 48;
  right: 8;
  justify-content: center;
`;

export {
  Wrap, Label, FieldWrap, Field,
  FieldError, FieldErrorClose, FieldErrorMessage, ActionButtonWrapper,
};
