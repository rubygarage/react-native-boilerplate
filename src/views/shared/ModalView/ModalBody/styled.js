import styled from 'styled-components/native';

const Wrap = styled.View`
  padding-top: 24;
  padding-right: 24;
  padding-left: 24;
`;

// eslint-disable-next-line import/prefer-default-export
export { Wrap };
