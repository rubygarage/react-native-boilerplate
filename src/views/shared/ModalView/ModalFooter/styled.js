import styled from 'styled-components/native';

const Wrap = styled.View`
  flex-shrink: 0;
  flex-direction: row;
  justify-content: flex-end;
  padding: 16px;
`;

// eslint-disable-next-line import/prefer-default-export
export { Wrap };
