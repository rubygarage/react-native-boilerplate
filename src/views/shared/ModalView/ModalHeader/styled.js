import styled from 'styled-components/native';

const Wrap = styled.View`
  position: relative;
  flex-direction: row;
  align-items: center;
  flex-shrink: 0;
  padding: 16px 24px;
  border-bottom-color: rgba(82,97,115, 0.12);
  border-bottom-width: 1;
  border-style: solid;
`;

const Title = styled.Text`
  font-size: 16;
  line-height: 24;
  letter-spacing: -0.2;
  font-weight: bold;
  color: ${({ theme }) => theme.colors.black};
`;

const Close = styled.TouchableOpacity`
  flex-shrink: 0;
  margin-right: 16;
`;

export { Wrap, Title, Close };
