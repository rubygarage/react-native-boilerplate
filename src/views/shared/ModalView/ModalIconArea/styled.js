import styled from 'styled-components/native';

import IconView from 'views/shared/Icon';

const Wrap = styled.View`
  justify-content: center;
  align-items: center;
  margin-top: 24;
  margin-bottom: 24;
`;

const Icon = styled(IconView)`
  margin-bottom: 16;
`;

const Text = styled.Text`
  font-size: 14;
  line-height: 20;
  color: ${({ theme }) => theme.colors.black};
`;

export { Wrap, Icon, Text };
