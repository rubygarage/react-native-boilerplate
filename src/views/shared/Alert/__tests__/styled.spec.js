import React from 'react';

import renderStyled from 'utils/testHelpers/renderStyled';
import * as S from '../styled';

const color = 'white';

describe('Wrapper styled components', () => {
  describe('S.Wrapper matches snapshot', () => {
    const defaultProps = { color };

    it('default', () => {
      const wrapper = renderStyled(<S.Wrapper {...defaultProps} />);
      expect(wrapper).toMatchSnapshot();
    });

    it('closeable', () => {
      const props = { ...defaultProps, onCloseClick: jest.fn() };
      const wrapper = renderStyled(<S.Wrapper {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('S.IconWrap matches snapshot', () => {
    it('default', () => {
      const wrapper = renderStyled(<S.IconWrap />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('S.ContentArea matches snapshot', () => {
    it('default', () => {
      const wrapper = renderStyled(<S.ContentArea />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('S.MessageTitle matches snapshot', () => {
    it('default', () => {
      const wrapper = renderStyled(<S.MessageTitle color={color} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('S.MessageText matches snapshot', () => {
    it('default', () => {
      const wrapper = renderStyled(<S.MessageText color={color} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('S.ActionButton matches snapshot', () => {
    it('default', () => {
      const wrapper = renderStyled(<S.ActionButton color={color} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('S.ActionButtonText matches snapshot', () => {
    it('default', () => {
      const wrapper = renderStyled(<S.ActionButtonText />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('S.ActionIcon matches snapshot', () => {
    it('default', () => {
      const wrapper = renderStyled(<S.ActionIcon />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('S.CloseButton matches snapshot', () => {
    it('default', () => {
      const wrapper = renderStyled(<S.CloseButton />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
