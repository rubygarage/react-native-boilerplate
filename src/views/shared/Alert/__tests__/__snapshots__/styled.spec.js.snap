// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper styled components S.ActionButton matches snapshot default 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderRadius": 4,
      "flexDirection": "row",
      "height": 36,
      "marginTop": 8,
      "maxWidth": 200,
      "opacity": 1,
      "paddingLeft": 12,
      "paddingRight": 12,
    }
  }
/>
`;

exports[`Wrapper styled components S.ActionButtonText matches snapshot default 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#ffffff",
        "fontSize": 14,
        "fontWeight": "bold",
        "letterSpacing": -0.2,
        "lineHeight": 20,
      },
    ]
  }
/>
`;

exports[`Wrapper styled components S.ActionIcon matches snapshot default 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      Array [
        Object {
          "marginRight": 24,
        },
      ],
      Object {
        "fontFamily": "icomoon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`Wrapper styled components S.CloseButton matches snapshot default 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "color": "#ffffff",
      "opacity": 1,
      "position": "absolute",
      "right": 12,
      "top": 12,
    }
  }
/>
`;

exports[`Wrapper styled components S.ContentArea matches snapshot default 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "display": "flex",
        "flexGrow": 1,
        "paddingRight": 36,
      },
    ]
  }
/>
`;

exports[`Wrapper styled components S.IconWrap matches snapshot default 1`] = `
<View
  style={
    Array [
      Object {
        "flexShrink": 0,
        "marginRight": 12,
      },
    ]
  }
/>
`;

exports[`Wrapper styled components S.MessageText matches snapshot default 1`] = `
<Text
  color="white"
  style={
    Array [
      Object {
        "color": "white",
        "fontSize": 14,
        "lineHeight": 20,
        "marginBottom": 2,
      },
    ]
  }
/>
`;

exports[`Wrapper styled components S.MessageTitle matches snapshot default 1`] = `
<Text
  color="white"
  style={
    Array [
      Object {
        "color": "white",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "fontSize": 14,
        "fontWeight": "bold",
        "lineHeight": 20,
        "marginTop": 2,
      },
    ]
  }
/>
`;

exports[`Wrapper styled components S.Wrapper matches snapshot closeable 1`] = `
<View
  color="white"
  onCloseClick={[MockFunction]}
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderRadius": 4,
        "flexDirection": "row",
        "marginBottom": 8,
        "paddingBottom": 12,
        "paddingLeft": 12,
        "paddingRight": 48,
        "paddingTop": 12,
        "position": "relative",
      },
    ]
  }
/>
`;

exports[`Wrapper styled components S.Wrapper matches snapshot default 1`] = `
<View
  color="white"
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderRadius": 4,
        "flexDirection": "row",
        "marginBottom": 8,
        "paddingBottom": 12,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 12,
        "position": "relative",
      },
    ]
  }
/>
`;
